src/IO/ReadBufferFromS3.cpp

```diff
@@ -43,7 +43,6 @@ ReadBufferFromS3::ReadBufferFromS3(
 
 bool ReadBufferFromS3::nextImpl()
 {
-    Stopwatch watch;
```

Comment:
We should probably count the backoff somewhere as well... otherwise it is going to look as some mysterious delays during reading that are not counted anywhere.

Reply:
Actually we already have mysterious delays when writing to S3, and in cases when we have errors while connecting to S3 for reading. These backoffs are handled by S3 client. May be we need to create an issue for that.

Reply:
@akuzm This is a metric which shall not account waits between attempts, by its definition: `Time of GET and HEAD requests to S3 storage.` It is not of a level of S3 request, which is retryable, but rather about underlying HTTP requests.